src/RaceHandler.js

```diff
@@ -0,0 +1,50 @@
+import { Console, Random } from '@woowacourse/mission-utils';
+
+class RaceHandler {
+    onStartLine(carNames) {
+        return Object.fromEntries(carNames.map(carName => [carName, '']));
+    }
+
+    doRace(raceBoard, carNames) {
+        carNames.forEach(carName => {
+            const hasMoved = Random.pickNumberInRange(0, 9);
+            if (hasMoved >= 4) {
+                raceBoard[carName] += '-';
+            }
+        });
+    }
+
+    showRaceBoard(raceBoard) {
+        Object.entries(raceBoard).forEach(([carName, moves]) => {
+            Console.print(`${carName} : ${moves}`);
+        });
+        Console.print('');
+    }
+
+    showWinner(raceBoard, carNames) {
+        let winners = [];
+        let maxMoves = 0;
+
+        carNames.forEach(carName => {
+            const moves = raceBoard[carName].length;
+            if (moves > maxMoves) {
+                maxMoves = moves;
+                winners = [carName];
+            } else if (moves === maxMoves) {
+                winners.push(carName);
+            }
+        });
+
+        this.validateWinner(maxMoves);
+        const result = winners.join(', ');
+        Console.print(`최종 우승자 : ${result}`);
+    }
```

Comment:
showWinner라는 함수가 winner를 '계산'하고 '보여주는'역할을 하고 있는 것 같아요. 두 역할을 분리해주는 건 어떨까요?

Reply:
```suggestion
    showWinner(raceBoard, carNames) {
        let winners = [];
        let maxMoves = 0;
        
        let maxMoves = Math.max(...carNames.map(carName => raceBoard[carName].length));
        carNames.filter(carName => raceBoard[carName].length === maxMoves);
        this.validateWinner(maxMoves);
        const result = winners.join(', ');
        Console.print(`최종 우승자 : ${result}`);
    }
```
이렇게도 나타낼 수 있을 것 같아요!:)

Reply:
`carNames.filter(carName => raceBoard[carName].length === maxMoves);`
요거는
`winners = carNames.filter(carName => raceBoard[carName].length === maxMoves);`
이런 식으로 담아서 쓰는 건가요?

검색해보니 filter는 조건을 만족하는 새로운 배열을 반환한다구 나와서
변수에 할당해야 쓸 수 있는 게 아닌가 여쭤봅니다!

Reply:
앗 맞아요!말쓴하신것 처럼 winners에 담아서 사용해야해요 제가 제안한 코드에 문제가 좀 있었네요!!:)